deploifai/cli/application/create.py

```diff
@@ -0,0 +1,316 @@
+import docker
+from time import sleep
+from InquirerPy import prompt
+from click_spinner import spinner
+
+from deploifai.cli.api import DeploifaiAPIError
+from deploifai.cli.context import pass_deploifai_context_obj, DeploifaiContextObj, is_authenticated, project_found, local_config
+
+
+@click.command()
+@click.argument("name")
+@click.option(
+    "--env",
+    help="Environment variables for the container (e.g. --env path/to/env OR --env VAR1=VAL1 --env VAR2=VAL2)",
+    multiple=True,
+)
+@click.option("--port", "-p", help="Port of container")
+@pass_deploifai_context_obj
+@is_authenticated
+@project_found
+def create(context: DeploifaiContextObj, name: str, env: tuple, port: int):
+    """
+    Create or update a new deployment
+    """
+    # Query required data
+    project_id = context.local_config['PROJECT']['id']
+    workspace = context.global_config['WORKSPACE']['username']
+
+    # Check if name is taken
+    where_application = {
+        "name": {"equals": name},
+        "project": {"is": {"id": {"equals": project_id}}},
+        "status": {"not": {"equals": "DESTROY_SUCCESS"}}
+    }
+    try:
+        applications = context.api.get_applications(workspace=workspace, where_application=where_application)
+        if len(applications) > 0:
+            click.secho("Application name already exists", fg="red")
+            raise click.Abort()
+    except DeploifaiAPIError:
+        click.secho("Error while querying applications", fg="red")
+        raise click.Abort()
+
+    # Check if name is valid
+    if len(name) < 4 or len(name) > 30 or not re.match(r"^[a-z0-9-]+$", name):
+        click.secho("Application name must contain only lowercase alphanumeric characters, and -, not contain spaces, "
+                    "and must be between 4 and 30 characters long", fg="red")
+        raise click.Abort()
+
+    # Check if port is valid
+    if port:
+        try:
+            port = int(port)
+        except ValueError:
+            click.secho("Port must be an integer", fg="red")
+            raise click.Abort()
+        if port < 0 or port > 65535:
+            click.secho("Port must be between 0 and 65535", fg="red")
+            raise click.Abort()
+
+    # Check if env is valid
+    if env:
+        # Check if env is a file
+        if len(env) == 1 and not re.match(r"^.+=.+$", env[0]):
+            if not os.path.exists(env[0]):
+                click.secho("Environment file does not exist", fg="red")
+                raise click.Abort()
+            with open(env[0], 'r') as f:
+                env = f.read().splitlines()
+        for env_var in env:
+            if not re.match(r"^.+=.+$", env_var):
+                click.secho(f"Invalid environment variable: {env_var}\n"
+                            f"Environment variables must be in the format of KEY=VALUE", fg="red")
+                raise click.Abort()
+    # Format env
+    formatted_env = []
+    for env_var in env:
+        key, val = env_var.split("=", maxsplit=1)
```

Comment:
Catch python `ValueError` when `env_var` fails to unpack into a key-value pair, this would indicate that this `env_var` is faulty.

Print this `env_var` for the user, and abort the command